arbos/arbosState/arbosstate.go

```diff
@@ -296,7 +296,16 @@ func (state *ArbosState) UpgradeArbosVersion(
 					ErrFatalNodeOutOfDate,
 				)
 			}
-			// no state changes needed
+			// Update the PerBatchGasCost to a more accurate value compared to the old v6 default.
+			ensure(state.l1PricingState.SetPerBatchGasCost(l1pricing.InitialPerBatchGasCostV12))
+
+			// We had mistakenly initialized AmortizedCostCapBips to math.MaxUint64 in older versions,
+			// but the correct value to disable the amortization cap is 0.
+			oldAmortizationCap, err := state.l1PricingState.AmortizedCostCapBips()
+			ensure(err)
+			if oldAmortizationCap == math.MaxUint64 {
+				ensure(state.l1PricingState.SetAmortizedCostCapBips(0))
```

Comment:
We could alternatively change the amortization cap code to recognize MaxUint64 as disabling the amortization cap, but that's a trickier value and I doubt we'd ever want a true amortization cap of zero.

Reply:
updating the cap to 0 makes sense, since thats the value we check against for versions >= 3 in l1pricing code